store/mockstore/mocktikv/mvcc_leveldb.go

```diff
@@ -1273,3 +1273,95 @@ func (mvcc *MVCCLevelDB) doRawDeleteRange(startKey, endKey []byte) error {
 
 	return mvcc.db.Write(batch, nil)
 }
+
+// MvccGetByStartTS implements the MVCCDebugger interface.
+func (mvcc *MVCCLevelDB) MvccGetByStartTS(startKey, endKey []byte, starTS uint64) (*kvrpcpb.MvccInfo, []byte) {
+	mvcc.mu.RLock()
+	defer mvcc.mu.RUnlock()
+
+	pairs := mvcc.ReverseScan(nil, nil, 1, starTS, kvrpcpb.IsolationLevel_RC)
+	if len(pairs) != 1 {
+		return nil, nil
+	}
+
+	key := pairs[0].Key
+	return mvcc.MvccGetByKey(key), key
+}
+
+var valueTypeOpMap = map[mvccValueType]kvrpcpb.Op{
+	typePut:      kvrpcpb.Op_Put,
+	typeDelete:   kvrpcpb.Op_Del,
+	typeRollback: kvrpcpb.Op_Rollback,
+}
+
+// MvccGetByKey implements the MVCCDebugger interface.
+func (mvcc *MVCCLevelDB) MvccGetByKey(key []byte) *kvrpcpb.MvccInfo {
+	mvcc.mu.RLock()
+	defer mvcc.mu.RUnlock()
+
+	info := &kvrpcpb.MvccInfo{}
+
+	startKey := mvccEncode(key, lockVer)
+	iter := newIterator(mvcc.db, &util.Range{
+		Start: startKey,
+	})
+	defer iter.Release()
+
+	dec1 := lockDecoder{expectKey: key}
+	ok, err := dec1.Decode(iter)
+	if err != nil {
+		return nil
+	}
+	if ok {
+		var shortValue []byte
+		if isShortValue(dec1.lock.value) {
+			shortValue = dec1.lock.value
+		}
+		info.Lock = &kvrpcpb.MvccLock{
+			Type:       dec1.lock.op,
+			StartTs:    dec1.lock.startTS,
+			Primary:    dec1.lock.primary,
+			ShortValue: shortValue,
+		}
+	}
+
+	dec2 := valueDecoder{expectKey: key}
+	var writes []*kvrpcpb.MvccWrite
+	var values []*kvrpcpb.MvccValue
+	for iter.Valid() {
+		ok, err := dec2.Decode(iter)
+		if err != nil {
+			return nil
+		}
+		if !ok {
+			iter.Next()
+			continue
```

Comment:
should break?
Otherwise, the iter will scan to end value?

Reply:
https://github.com/pingcap/tidb/blob/a4d6f8863617f2179d9a616f7e8a9848ad4e8f3e/store/mockstore/mocktikv/mvcc_leveldb.go#L223-L225
No, the `valueDecoder.Decode()` will return false if current iter key not match the excepted key.
So we need traverse the entire iteratror, and skip these unmatched keys.

Reply:
> So we need traverse the entire iteratror, and skip these unmatched keys.

I'm confused about that. It actually scans to the last one if you write code like that.

Reply:
We need to scan all the versions of a key, but not all keys.